core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts

```diff
@@ -643,6 +684,258 @@ describe('DotEmaShellComponent', () => {
             });
         });
 
+        describe('language checking', () => {
+            it('should not trigger the confirmation service if the page is translated to the current language', () => {
+                const confirmationServiceSpy = jest.spyOn(confirmationService, 'confirm');
```

Comment:
@zJaaal  I suggest avoiding to create spies manually per each test, instead if you send in `componentProviders` the service like this `mockProvider(ConfirmationService)` automatically all methods can be spied therefore avoiding to create a manual spy.

Reply:
This didn't work. I just need to do that for that method so I just do it once in the `beforeEach`